community/kernel/src/main/java/org/neo4j/kernel/impl/locking/Locks.java

```diff
@@ -107,6 +109,9 @@ public Factory( String key, String... altKeys )
         /** Release all locks. */
         void releaseAll();
 
+        /** Associate with transaction. */
+        void setTx( Transaction tx );
```

Comment:
I'm cringing when looking at this setter. Why don't you make this association when creating/binding the client instead?